htdocs/projet/tasks/time.php

```diff
@@ -268,18 +271,20 @@
 				$object->timespent_date = dol_mktime(12, 0, 0, GETPOST("timelinemonth"), GETPOST("timelineday"), GETPOST("timelineyear"));
 			}
 			$object->timespent_fk_user = GETPOST("userid_line", 'int');
+			$object->timespent_fk_product = GETPOST("fk_product", 'int');
 
 			$result = 0;
 			if (in_array($object->timespent_fk_user, $childids) || $user->rights->projet->all->creer) {
 				$result = $object->addTimeSpent($user);
+				if ($result >= 0) {
+					setEventMessages($langs->trans("RecordSaved"), null, 'mesgs');
+				} else {
+					setEventMessages($langs->trans($object->error), null, 'errors');
+					$error++;
+				}
 			}
 
-			if ($result >= 0) {
-				setEventMessages($langs->trans("RecordSaved"), null, 'mesgs');
-			} else {
-				setEventMessages($langs->trans($object->error), null, 'errors');
-				$error++;
-			}
+			$result = $object->addTimeSpent($user);
```

Comment:
It seems this line is executed twice. 
It is already executed before (with a condition on authorization to do it). I think we must remove this line (that is done without authorization control). Can you confirm ?